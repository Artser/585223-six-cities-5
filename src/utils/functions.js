export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
export const MAX_REVIEWS_LENGTH = 10;
export const FavoriteStatus = {
  PLUS: 1,
  MINUS: 0
};
export const getIdFromCityName = (cityName) => {
  switch (cityName) {
    case `Amsterdam`: return 4;
    case `Hamburg`: return 5;
    case `Paris`: return 1;
    case `Cologne`: return 2;
    case `Dusseldorf`: return 6;
    case `Brussels`: return 3;
    default:
      return 0;
  }

};
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};
export const PagePath = {
  MAIN: `/`,
  FAVORITE: `/favorites`,
  LOGIN: `/login`,
  OFFER: `/offer/`
};
export const SORT_TYPES = {
  POPULAR: `Popular`,
  TO_HIGH_PRICE: `Price: low to high`,
  TO_LOW_PRICE: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};

export const getSortedOffers = (offers, activeSort) => {
  switch (activeSort) {
    case SORT_TYPES.POPULAR:
      return offers.slice();
    case SORT_TYPES.TO_HIGH_PRICE:
      return offers.slice().sort((a, b) => a.price - b.price);
    case SORT_TYPES.TO_LOW_PRICE:
      return offers.slice().sort((a, b) => b.price - a.price);
    case SORT_TYPES.TOP_RATED:
      return offers.slice().sort((a, b) => b.rating - a.rating);
    default:
      return offers.slice();
  }
};
export const getRating = (rating) => {
  const roundRating = Math.round(rating);
  return roundRating > 0 && roundRating <= 5 ? `${roundRating * 20}%` : false;
};
