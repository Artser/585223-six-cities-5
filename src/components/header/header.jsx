import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { AuthorizationStatus, PagePath } from "../../utils/functions";
import { connect } from "react-redux";

export const Header = (props) => {
  const { authorizationStatus, authInfo } = props;
  return <header className="header">
    <div className="container">
      <div className="header__wrapper">
        <div className="header__left">
          <Link to="/" className="header__logo-link header__logo-link--active">
            <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
          </Link>
        </div>
        <nav className="header__nav">
          <ul className="header__nav-list">
            <li className="header__nav-item user">
              <span className="header__nav-link header__nav-link--profile">
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                >
                </div>
                {
                  authorizationStatus === AuthorizationStatus.AUTH ?
                    <Link className="header__user-name user__name" to={PagePath.FAVOURITE}>
                      {authInfo ? authInfo.email : ``}
                    </Link>
                    : <Link className="header__login" to={PagePath.LOGIN}>Sign in</Link>

                }
              </span>
            </li>

          </ul>
        </nav>
      </div>
    </div>
  </header>;

};
Header.propTypes = {

  authorizationStatus: PropTypes.string,
  authInfo: PropTypes.object,

};
const mapStateToProps = (state) => {


  return {
    authorizationStatus: state.authorizationStatus,
    authInfo: state.authInfo,

  };
};
export default connect(mapStateToProps)(Header);
