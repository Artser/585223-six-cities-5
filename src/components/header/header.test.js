import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { AuthorizationStatus } from '../../utils/functions';
import {Header} from './header';

it(`Header render correctly with active feature`, () => {
  const tree = renderer.create(<MemoryRouter>
  <Header
    authInfo={false}
    authorizationStatus={AuthorizationStatus.NO_AUTH}
  />
  </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Header render correctly without active feature`, () => {
  const tree = renderer.create(<MemoryRouter>

  <Header
    authInfo={`Dusseldorf@mail.ru`}
    authorizationStatus={AuthorizationStatus.AUTH}
  />
  </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
