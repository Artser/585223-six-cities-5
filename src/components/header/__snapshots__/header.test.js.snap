// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly with active feature 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <span
              className="header__nav-link header__nav-link--profile"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <a
                className="header__login"
                href="/login"
                onClick={[Function]}
              >
                Sign in
              </a>
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header render correctly without active feature 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link header__logo-link--active"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <span
              className="header__nav-link header__nav-link--profile"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
              />
              <a
                className="header__user-name user__name"
                href="/favorites"
                onClick={[Function]}
              />
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
